src/StreamJsonRpc/Reflection/MessageFormatterProgressTracker.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc.Reflection
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using Microsoft;
+    using Microsoft.VisualStudio.Threading;
+    using Newtonsoft.Json.Linq;
+
+    /// <summary>
+    /// Class containing useful methods to help message formatters implement support for <see cref="IProgress{T}"/>.
+    /// </summary>
+    public class MessageFormatterProgressTracker
+    {
+        /// <summary>
+        /// Dictionary used to map progress id token to its corresponding <see cref="ProgressParamInformation" /> instance containing the progress object and the necessary fields to report the results.
+        /// </summary>
+        public readonly Dictionary<long, ProgressParamInformation> ProgressMap = new Dictionary<long, ProgressParamInformation>();
+
+        /// <summary>
+        /// Special method name for progress notification.
+        /// </summary>
+        internal const string ProgressRequestSpecialMethod = "$/progress";
```

Comment:
>internal [](start = 8, length = 8)

Why are all these members internal? Given you use this helper class from the test assembly, I'm guessing these needn't be accessed from outside this class. Can we make them all private? 
Or if some of them must be used from outside the class, should they not be made public so that formatters implemented external to this library can use them?

Reply:
Fixed, made them public or private according to how we use them.